extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/MSQWorkerTask.java

```diff
@@ -185,4 +189,22 @@ public int hashCode()
   {
     return Objects.hash(super.hashCode(), controllerTaskId, workerNumber, retryCount, worker);
   }
+
+  @Override
+  public LookupLoadingSpec getLookupLoadingSpec()
+  {
+    if (!getContext().containsKey(PlannerContext.CTX_LOOKUP_LOADING_MODE) || getContext().get(PlannerContext.CTX_LOOKUP_LOADING_MODE) == null) {
+      return LookupLoadingSpec.ALL;
+    }
+
+    String lookupLoadingMode = getContext().get(PlannerContext.CTX_LOOKUP_LOADING_MODE).toString();
+    if (lookupLoadingMode.equals(LookupLoadingSpec.Mode.NONE.toString())) {
+      return LookupLoadingSpec.NONE;
+    } else if (lookupLoadingMode.equals(LookupLoadingSpec.Mode.ONLY_REQUIRED.toString())) {
+      List<String> lookupsToLoad = (List<String>) getContext().get(PlannerContext.CTX_LOOKUPS_TO_LOAD);
```

Comment:
There should be some validation here to check if `lookupsToLoad` is null or empty.